CMakeLists.txt

```diff
@@ -398,6 +398,10 @@ if (ENABLE_CLANG_FORMAT)
   set(CMAKE_CXX_CLANG_TIDY ${CLANG_TIDY_EXE} "${CLANG_TIDY_FLAGS}")
 endif()
 
+add_custom_target(reformat
+    COMMAND find ${CMAKE_SOURCE_DIR} -iname '*.h' -o -iname '*.cpp' -exec python3 ${CMAKE_SOURCE_DIR}/Scripts/clang-format.py -i \\{\\} \\\\;
```

Comment:
Could you limit this to our actual source directories?

I'd prefer not having this command make assumptions about what name the build folder has (I actually use multiple) or what other files are in the source folder.

Instead of `find`, maybe `git ls-tree` would be a better tool to use here?

Reply:
Ah, `git ls-files` is an even better match, since it has a `-c` toggle with which the index is included as well.

Reply:
yes - git ls-files is definitely better, but why is the difference so large between what you reported on discord and my reformatting? Let me look at it.

Reply:
I have updated this now. I was a bit worried about putting a bash script in the CMakeLists command so I moved that to a `reformat.sh` script. What do you think?